### PR TITLE
Adding parser for Kosovo

### PR DESCRIPTION
**Pull request by dchaplinsky**
Fixing

### CODE REVIEW

File: datasets/xk/companies/crawler.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        founders = resp[0].get("pronaret", [])
+        representatives = resp[0].get("perfaqesuesit", [])
+
+        company_type = company.get("LlojiBiznesit")
+        if company_type not in COMPANY_TYPES:
+            context.log.warning("Unknown company type: ", company_type)
+
+        status = company.get("StatusiARBK")
+        if status not in STATUSES:
+            context.log.warning("Unknown status: ", status)
```
Comment (jbothma) [inline datasets/xk/companies/crawler.py:161]: I'm seeing unknown status warnings, but I think there's no need to put maintenance effort into these for now. I'd say just stick the Albanian in there.

File: datasets/xk/companies/crawler.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+                entity.add("currency", capital["currency"])
+
+        if company.get("Adresa") or company.get("Vendi") or company.get("Komuna"):
+            address = h.make_address(
+                context,
+                street=company.pop("Adresa"),
+                city=company.pop("Vendi"),
+                country_code="xk",
+                # Should it be county or region?
+                state=company.pop("Komuna"),
```
Comment (jbothma) [inline datasets/xk/companies/crawler.py:204]: state looks good since it's the only sub-national division

File: datasets/xk/companies/crawler.py [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+        company_type = company.get("LlojiBiznesit")
+        if company_type not in COMPANY_TYPES:
+            context.log.warning("Unknown company type: ", company_type)
+
+        status = company.get("StatusiARBK")
+        if status not in STATUSES:
+            context.log.warning("Unknown status: ", status)
+
+        entity = context.make("Company")
+        entity.id = context.make_id("XKCompany", company_id)
```
Comment (jbothma) [inline datasets/xk/companies/crawler.py:164]: I'm a touch nervous of using these non-permalink IDs for the entity ID - I think it'd be a good ida to do something like
if  NUI:
  id = make_slug(nui, get(nui))
elif NumriBiznesit
    id = make_slug(NumriBiznesit, get(NumriBiznesit))
elif NumriFiskal:
    id = ...
else:
    id = make_id(..., company_id)

that way if they redesign the website for fun, we can likely reconstruct most of the old IDs.

File: datasets/xk/companies/crawler.py [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+                state=company.pop("Komuna"),
+            )
+            h.apply_address(context, entity, address)
+
+        if company.get("NumriPunetoreve"):
+            entity.add(
+                "notes",
+                f'Number of employees: {company.pop("NumriPunetoreve")}',
+                lang="eng",
+            )
```
Comment (jbothma) [inline datasets/xk/companies/crawler.py:213]: I think you can drop this for now - it's already going to be a LOT of data and I don't think this is a priority.

**Review by jbothma — COMMENTED** [review]
Super nice, thank you! just a few small comments and I wanna confirm adding the cryptography library with Friedrich then I think we can merge

**Comment by dchaplinsky** [comment]
@jbothma  in your hands now

File: datasets/xk/companies/crawler.py [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+                "XKdirector", entity.id, rep["Emri"], rep["Mbiemri"]
+            )
+            director.add("name", f'{rep["Emri"]} {rep["Mbiemri"]}', lang="sqi")
+            context.emit(director)
+
+            rel = context.make("Directorship")
+            rel.id = context.make_id("XKDirectorship", entity.id, director.id)
+            rel.add("role", rep["Pozita"], lang="sqi")
+
+            # JD: Should it be description or notes or status or summary?
```
Comment (jbothma) [inline datasets/xk/companies/crawler.py:258]:
```suggestion
```

File: datasets/xk/companies/crawler.py [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+            context.emit(founder)
+
+            own = context.make("Ownership")
+            own.id = context.make_id("XKOwnership", entity.id, founder.id)
+            own.add("asset", entity.id)
+            own.add("owner", founder.id)
+
+            own.add("ownershipType", "Founder", lang="eng")
+
+            # JD: Not sure where to map shares and capital_contracted
```
Comment (jbothma) [inline datasets/xk/companies/crawler.py:241]:
```suggestion
```

File: datasets/xk/companies/crawler.py (hunk 1 above)
Comment (jbothma) [reply datasets/xk/companies/crawler.py]: Let's use the albanian

**Pull request merged**